### ISSUE TITLE
Difficulties setting up dredd-hooks-rust in existing project

### ISSUE THREAD

**Issue by stefan-vatov** [hobofan/dredd-hooks-rust#7]
Hey,

First of all, thanks for setting this up!

I found myself here after following https://dredd.org/en/latest/hooks/rust.html#hooks-rust and trying add it to an exiting project. Following that + the `Readme` here, I've been trying to setup dredd hooks into an existing project, but I'm running into some issues:

1. Running `cargo install dredd-hooks` ends with

```bash
.....
Compiling tokio-proto v0.1.1
   Compiling dredd-hooks v0.3.0
    Finished release [optimized] target(s) in 9.51s
error: no binaries are available for install using the selected features
```

Which I think eventually leads to problems when trying to run `cargo dredd`:

```bash
cargo dredd <openapicontract.yml> http://127.0.0.1:8000
error: no such subcommand: `dredd`

	Did you mean `tree`?
```

2. I've added the sections to`Cargo.toml` as describe in the `Readme` here, but I'm not sure how to build a binary from just the hooks file. (could be I'm still farily new to Rust)

```toml
[[test]]
name = "dredd_test_hooks"
path = "tests/dredd/hooks.rs"
test = false
harness = false

[package.metadata.dredd_hooks]
hook_targets = ["dredd_test_hooks"]
```

The hooks file is the one from the `examples` folder here just to help the initial setup.

----

Edit: Forgot to mention, while I see that the last commits are from 2017, _I assume_ it works as expected and didn't need changes 😄 

Also trying to install via:

`cargo install dredd-hooks --features binaries` 

fails with

```bash
......
error: failed to run custom build command for `openssl v0.9.24`
......
......
 --- stderr
  thread 'main' panicked at 'Unable to detect OpenSSL version'
......
```

**Comment by hobofan** [comment]
`cargo install dredd-hooks --features binaries` does sound like the right command for installing.

Your last error looks like the normal error you would get if you don't have openssl installed. You can install it or specify it via various different options: https://docs.rs/openssl/latest/openssl/#automatic

I haven't used the crate myself in 5 years, so everything that's not written down in the repo I've forgotten by now, sorry 😅

**Comment by stefan-vatov** [comment]
Thanks for the quick reply!

Yeah, I managed to get it to build & install by updating `cargo` and fixing 1 breaking change. Running it breaks when it tries to read the `Cargo.lock`. I suspect a full update of all packages is needed to get it working, but that brings in a bunch of other breaking changes 🤣

No worries, I got the contract tests running using the JS hooks. Could be worth pinging the dredd people, letting them know it doesn't work, since it's in the official docs 😄
